src/NLog.Targets.Syslog/AsyncLogger.cs

```diff
@@ -109,36 +112,20 @@ private Task ProcessQueueAsync(MessageBuilder messageBuilder, TaskCompletionSour
 
         private void Enqueue(AsyncLogEventInfo asyncLogEventInfo, int timeout)
         {
-            queue.TryAdd(asyncLogEventInfo, timeout, token);
+            if (!queue.TryAdd(asyncLogEventInfo, timeout, token))
+                asyncLogEventInfo.Continuation(new InvalidOperationException("AsyncLogger Queue Full"));
```

Comment:
Well to me `TryAdd` means it can return false, which should be handled.

Reply:
Enque is called by [`Throttling.Apply`](https://github.com/luigiberrettini/NLog.Targets.Syslog/blob/master/src/NLog.Targets.Syslog/Policies/Throttling.cs#L36) in the [`Log` method](https://github.com/luigiberrettini/NLog.Targets.Syslog/blob/b9df517e502cef466eab35d9f0e156a1ca0bd43a/src/NLog.Targets.Syslog/AsyncLogger.cs#L42) with a timeout that can be infinite (it will only return true) or a value after which the message should be discarded, therefore your change would break the desired behavior that is currently in place

Reply:
When a NLog LogEvent is discarded then its Continuation should be called anyway. But I will remove this code.

Reply:
In that case the code should be kept if possible without exceptions